### ISSUE TITLE
When connecting with endpoints, the application factory used is neither started nor stopped, and irrelevant start/stop messages are logged instead

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5278]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5278|
|Type|defect|
|Created|2011-09-27 20:42:34Z|
|Branch|https://github.com/twisted/twisted/tree/startstop-messages-5278|

Instead, it reflects an endpoints implementation detail:

```
2011-09-27 16:40:11-0400 [-] Starting factory <twisted.internet.endpoints._WrappingFactory instance at 0x8a03eac>
2011-09-27 16:40:24-0400 [_WrappingProtocol,client] Stopping factory <twisted.internet.endpoints._WrappingFactory instance at 0x8a03eac>
```

These log messages come from a client factory.  In the case of endpoints though, they come from the internal endpoints helper factory, and they're pretty much always meaningless to anyone reading the logs.  Further, since the endpoints helper factory doesn't bother to start or stop the actual application factory, that factory never logs any start or stop messages.

<details><summary>Searchable metadata</summary>

```
trac-id__5278 5278
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_startstop_messages_5278 branches/startstop-messages-5278
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1317156154000000 1317156154000000
changetime__1318333419000000 1318333419000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [32779]) Branching to 'startstop-messages-5278'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [32780]) Test the IListeningPort started-listening messages for TCP, TLS, UNIX, and make them nicer with respect to wrapping factories of various sorts

refs [#5278](https://github.com/twisted/twisted/issues/5278)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

So I got a little mixed up here.  The problem in the ticket summary was resolved already by [#5062](https://github.com/twisted/twisted/issues/5062).  However, the contents of the log message for this case are still not ideal - the `_WrappingFactory` shows up and the wrapped factory does not, because the wrapped factory never even gets started or stopped.  I think it should, so that's what I'll try to implement here.

Also, I got side-tracked and fixed some server-side stuff related to wrapping factories too.  Maybe that should be split off into another ticket, we'll see.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [32781]) Pass doStart and doStop through to the wrapped factory; disregard it otherwise in the endpoint wrapping factory, so as not to log useless messages about the endpoint factory starting and stopping.

refs [#5278](https://github.com/twisted/twisted/issues/5278)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [32782]) Make the logging tests pass on IOCP reactor as well

refs [#5278](https://github.com/twisted/twisted/issues/5278)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [32785]) Back out the port-related fixes, save them for another branch

refs [#5278](https://github.com/twisted/twisted/issues/5278)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Okay, the branch now only changes endpoints to start and stop the application-provided factory.  The endpoint factory itself no longer does anything itself at start and stop time (including logging messages).

Dunno if starting and stopping the wrapped factory was an intentional endpoints omission, but it seems like a bad thing to me.

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/startstop-messages-5278)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_8"></a>|@itamarst set owner to @exarkun|
|-|-|

Looks good, merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_9"></a>|@itamarst commented|
|-|-|

Oh, but, before you do, update the summary of this ticket to actually reflect the branch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun set status to `closed`|
|-|-|

(In [32812]) Merge startstop-messages-5278 

Author: exarkun
Reviewer: itamar
Fixes: [#5278](https://github.com/twisted/twisted/issues/5278)

Change stream client endpoints to start and stop the application factory
supplied to the `connect` method, rather than a different (internal,
implementation-specific) factory.
